test-network/scripts/deployCC.sh

```diff
@@ -1,255 +1,307 @@
-    set -x
-    peer lifecycle chaincode querycommitted --channelID $CHANNEL_NAME --name fabcar >&log.txt
-    res=$?
-    set +x
+	# we either get a successful response, or reach MAX RETRY
+	while [ $rc -ne 0 -a $COUNTER -lt $MAX_RETRY ]; do
+		sleep $DELAY
+		echo "Attempting to Query committed status on peer0.org${ORG}, Retry after $DELAY seconds."
+		set -x
+		peer lifecycle chaincode querycommitted --channelID $CHANNEL_NAME --name ${CC_NAME} >&log.txt
+		res=$?
+		set +x
 		test $res -eq 0 && VALUE=$(cat log.txt | grep -o '^Version: [0-9], Sequence: [0-9], Endorsement Plugin: escc, Validation Plugin: vscc')
-    test "$VALUE" = "$EXPECTED_RESULT" && let rc=0
+		test "$VALUE" = "$EXPECTED_RESULT" && let rc=0
 		COUNTER=$(expr $COUNTER + 1)
 	done
-  echo
-  cat log.txt
-  if test $rc -eq 0; then
-    echo "===================== Query chaincode definition successful on peer0.org${ORG} on channel '$CHANNEL_NAME' ===================== "
+	echo
+	cat log.txt
+	if test $rc -eq 0; then
+		echo "===================== Query chaincode definition successful on peer0.org${ORG} on channel '$CHANNEL_NAME' ===================== "
+		echo
+	else
+		echo
+		echo $'\e[1;31m'"!!!!!!!!!!!!!!! After $MAX_RETRY attempts, Query chaincode definition result on peer0.org${ORG} is INVALID !!!!!!!!!!!!!!!!"$'\e[0m'
 		echo
-  else
-    echo "!!!!!!!!!!!!!!! After $MAX_RETRY attempts, Query chaincode definition result on peer0.org${ORG} is INVALID !!!!!!!!!!!!!!!!"
-    echo
-    exit 1
-  fi
+		exit 1
+	fi
 }
 
 chaincodeInvokeInit() {
-  parsePeerConnectionParameters $@
-  res=$?
-  verifyResult $res "Invoke transaction failed on channel '$CHANNEL_NAME' due to uneven number of peer and org parameters "
-
-  # while 'peer chaincode' command can get the orderer endpoint from the
-  # peer (if join was successful), let's supply it directly as we know
-  # it using the "-o" option
-  set -x
-  peer chaincode invoke -o localhost:7050 --ordererTLSHostnameOverride orderer.example.com --tls --cafile $ORDERER_CA -C $CHANNEL_NAME -n fabcar $PEER_CONN_PARMS --isInit -c '{"function":"initLedger","Args":[]}' >&log.txt
-  res=$?
-  set +x
-  cat log.txt
-  verifyResult $res "Invoke execution on $PEERS failed "
-  echo "===================== Invoke transaction successful on $PEERS on channel '$CHANNEL_NAME' ===================== "
-  echo
+	parsePeerConnectionParameters $@
+	res=$?
+	verifyResult $res "Invoke transaction failed on channel '$CHANNEL_NAME' due to uneven number of peer and org parameters "
+
+	# while 'peer chaincode' command can get the orderer endpoint from the
+	# peer (if join was successful), let's supply it directly as we know
+	# it using the "-o" option
+	set -x
+	fcn_call='{"function":"'${CC_INIT_FCN}'","Args":[]}'
+	echo invoke fcn call:${fcn_call}
+	peer chaincode invoke -o localhost:7050 --ordererTLSHostnameOverride orderer.example.com --tls --cafile $ORDERER_CA -C $CHANNEL_NAME -n ${CC_NAME} $PEER_CONN_PARMS --isInit -c ${fcn_call} >&log.txt
```

Comment:
We're not using init-required and --isInit now, right?

Reply:
we are using both if the user defines an init function